experiments/browser_based_querying/src/hackernews/adapter.ts

```diff
@@ -71,25 +71,17 @@ const _itemPattern = /^https:\/\/news\.ycombinator\.com\/item\?id=(\d+)$/;
 const _userPattern = /^https:\/\/news\.ycombinator\.com\/user\?id=(.+)$/;
 
 function materializeWebsite(fetchPort: MessagePort, url: string): Vertex | null {
-  const itemMatch = url.match(_itemPattern);
-  let ret: any
-  if (itemMatch) {
+  let matcher: RegExpMatchArray | null = null;
+  let ret: any | null = { __typename: 'Website' };
```

Comment:
Can `ret` ever be `null` here?

Can `materializeItem` or `materializeUser` return `null`, e.g. if the item or user have been deleted? If so, `ret.url` on line 83 might be a problem.

It seems that either the declared type here or the assignment on line 83 is buggy.

Reply:
I fixed this in my code in the next commit, but really the code is scattered with direct materializeUser's which don't add .url so the fix is to just make them all defer back to materializeWebsite rather than directly calling materializeUser so the resulting object will have .url too, that will all be changed the next pr though, so I'm not overly worried.

Reply:
Actually this isn't a problem because of even more indirection in which the url property is only read on Webpage objects, so their .url is never needed anyway.

Reply:
I'm in favor of simplifying. It sounds like all that indirection makes stuff not obvious right now so feel free to tweak it to make it more maintainable and easier to understand.